examples/run_shimming.py

```diff
@@ -0,0 +1,197 @@
+"""ppg for obtaining performing shimming calibration
+=====================================================
+The following is used when the shims are replaced or moved in any way. When
+moved the calibration or settings for the z and y shims need to be recalculated
+for certainty. Here, the user can calculate the appropriate voltage setting for
+each shim for optimal signal. Note: The XEPR_API server must be running for
+this script. Normally this script is run with 4 scans 
+"""
+from pylab import ones_like, shape
+from pyspecdata import getDATADIR, r_, figlist_var
+import os, h5py
+import SpinCore_pp
+from SpinCore_pp.ppg import run_spin_echo
+from datetime import datetime
+from Instruments.XEPR_eth import xepr
+from Instruments import HP6623A, prologix_connection
+
+target_directory = getDATADIR(exp_type="ODNP_NMR_comp/test_equipment")
+# {{{ importing acquisition parameters
+config_dict = SpinCore_pp.configuration("active.ini")
+nPoints = int(config_dict["acq_time_ms"] * config_dict["SW_kHz"] + 0.5)
+config_dict["tau_us"] = (
+    2500.0 + config_dict["deadtime_us"]
+)  # optimal tau for NiSO4 sample used by AB
+# }}}
+# {{{filename
+date = datetime.now().strftime("%y%m%d")
+config_dict["date"] = date
+filename = f"{config_dict['date']}_{config_dict['chemical']}_shimming" + ".h5"
+file_nodename = "shimming_" + str(config_dict["echo_counter"])
+# }}}
+# {{{phase cycling steps
+ph1_cyc = r_[0, 1, 2, 3]
+ph2_cyc = r_[0, 2]
+nPhaseSteps = 8
+# }}}
+# channel that is currently being investigated
+test_ch = 3
+# channel that was previously investigated/has predetermined value that you want to set
+# set_ch : (ch #, voltage setting)
+set_ch = [(2, 0.1)]
```

Comment:
these will need to be edited depending on which channel you are testing (details in protocol)